components/Editor/index.tsx

```diff
@@ -271,26 +274,61 @@ const Editor = ({ readOnly = false }: Props) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [vmError])
 
+  const validateBytecode = (bytecode: string) => {
+    if (bytecode.length % 2 !== 0) {
+      log('There should be at least 2 characters per byte', 'error')
+      return
+    }
+    if (!isHex(bytecode)) {
+      log('Only hexadecimal characters are allowed', 'error')
+      return
+    }
+  }
+
+  const stripBytecode = (value: string) => {
+    return value
+      .replaceAll(/\/\/.*$/gm, '')
+      .replaceAll(/;.*$/gm, '')
+      .replaceAll(/#.*$/gm, '')
+      .replaceAll(/\s/gm, '')
+  }
+
   const handleCodeChange = (value: string) => {
     setCode(value)
     setCodeModified(true)
+
+    try {
+      if (codeType === CodeType.Bytecode) {
+        const cleanBytecode = stripBytecode(value)
+
+        if (timeOutId) {
+          clearTimeout(timeOutId)
+          setTimeOutId(undefined)
+        }
+        setTimeOutId(setTimeout(() => validateBytecode(cleanBytecode), 1000))
+
+        loadInstructions(cleanBytecode)
```

Comment:
You're calling `loadInstructions` without validating code.
Wouldn't it be better to call inside timeout and after passing validation?

Reply:
Theoretically yes, but from what I saw in `loadInstructions()` code - if it's invalid - it outputs "INVALID" opcode, which also gives some information about what you've typed is invalid. Otherwise the opcodes view would just not update, which at least for my taste isn't a pleasant flow.
But semantically, if loadInstructions could at some point in future crash the whole application - that might be a problem.
Similar case with calling it inside timeout - it will update with a delay, even if it's correct. Instant refresh is a killer feature here IMHO.

Reply:
I see, makes sense.